impeller/renderer/backend/gles/buffer_bindings_gles.cc

```diff
@@ -279,20 +279,19 @@ bool BufferBindingsGLES::BindUniformBuffer(const ProcTableGLES& gl,
     auto* buffer_data =
         reinterpret_cast<const GLfloat*>(buffer_ptr + member.offset);
 
-    std::vector<uint8_t> array_element_buffer;
-    if (element_count > 1) {
-      // When binding uniform arrays, the elements must be contiguous. Copy
-      // the uniforms to a temp buffer to eliminate any padding needed by the
-      // other backends.
-      array_element_buffer.resize(member.size * element_count);
+    // When binding uniform arrays, the elements must be contiguous. Copy
+    // the uniforms to a temp buffer to eliminate any padding needed by the
+    // other backends if the array elements have padding.
+    if (element_count > 1 && element_stride != member.size) {
```

Comment:
If we have an array of vec4 then no backends will insert padding and we don't have to memcpy the data to a temp buffer.